com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -995,7 +1003,12 @@ void InitializeActuators()
             var param = m_PolicyFactory.BrainParameters;
             m_VectorActuator = new VectorActuator(this, param.ActionSpec);
             m_ActuatorManager = new ActuatorManager(attachedActuators.Length + 1);
-            m_LegacyActionCache = new float[m_VectorActuator.TotalNumberOfActions()];
+#pragma warning disable 618
+            m_LegacyActionCache = new float[param.VectorActionSpaceType == SpaceType.Continuous
```

Comment:
I think the old logic (`m_VectorActuator.TotalNumberOfActions()]`) was OK, since the vectorActuator should be sized to support only the BrainParameters (not the other actuators on the agent).

I'd also be OK allocating these lazily where they're needed.

Reply:
I'll convert this back